### PR TITLE
QRImageWidget: Sizing and font fixups

### PR DESCRIPTION
**Pull request by luke-jr**
- Margin of QRCode should be constant, not change based on complexity of QRCode
- Text region should be sized based on height of actual text, not a constant guess
- Width of image should not be adjusted based on vertical text size/margins
- Adjusted constant QRCode and margin sizes to visually match what we had previously
- If text won't fit, grow up to 25% wider, or split across multiple lines

(Note: Out of scope for this PR, after #497, I think we should allow customising the font used here and default to the embedded monospace for better privacy)

### CODE REVIEW

**Review by promag — COMMENTED** [review]
Concept ACK, however, I'll try a different approach for the multi-line case.

**Comment by promag** [comment]
@luke-jr feel free to try the commit 01730333996c614dfff12d13037e6673be4c2da1 ([branch](https://github.com/promag/gui/tree/pr/506)) which avoids manual text wrap by using `Qt::TextWrapAnywhere` flag.

Another thing I noticed is that the default minimum font size in `calculateIdealFontSize` is too small.

**Comment by luke-jr** [comment]
```c++
        auto rect = fm.boundingRect(max_rect, Qt::AlignCenter, text);
        if (rect.width() > max_rect.width() * 5 / 4) {
```

Isn't this impossible? I find the unnecessary QRect usage hard to follow.

>Another thing I noticed is that the default minimum font size in calculateIdealFontSize is too small.

IMO out of scope for this PR

**Comment by promag** [comment]
> Isn't this impossible?

No because it doesn't wrap.

> I find the unnecessary QRect usage hard to follow.

And I think it's unnecessary to calculate lines and manually wrap the text.

**Comment by hebasto** [comment]
@luke-jr Screenshots "before" and "after" in the PR description could be useful for further discussion.

**Review by jarolrod — COMMENTED** [review]
Concept ACK, stylistic changes look fine to me.

| master | PR (rebased) |
| ------- | ------------- |
Screen Shot 2022-06-28 at 2 11 01 AM | Screen Shot 2022-06-28 at 2 12 19 AM |

**Review by hebasto — COMMENTED** [review]
Tested 4c5143d56c98d4f9f961059692f155bbe2248c44.

This PR effectively makes font size smaller that makes it harder to read.

- master branch:
image

- this PR:
Screenshot from 2022-07-19 15-20-36

**Review by pablomartin4btc — COMMENTED** [review]
Concept ACK, I agree with the improvements on achieving a more consistent adjustment of the QR code image size. However, it appears that there is still an issue with the font size, which makes the address in the picture difficult to read. Is there a way to improve the legibility of the address text?

**Review by pablomartin4btc — COMMENTED** [review]
tested ACK 4c5143d56c98d4f9f961059692f155bbe2248c44

- master:

Screenshot from 2023-09-17 18-52-45

- this PR (rebased):

Screenshot from 2023-09-18 16-41-03
